erpnext/stock/report/stock_balance/test_stock_balance.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import unicode_literals
+import frappe, unittest
+from erpnext.stock.report.stock_balance.stock_balance import execute
+from frappe.utils import now, add_to_date
+
+class TestStockBalance(unittest.TestCase):
+
+	def test_for_stock_balance_with_date_filter(self):
+		item_1 = get_item("Test_Mac_Book_Pro", 12345)
+		item_2 = get_item("Test_One_Plus", 6789)
+
+		cols, rows = execute(filters=frappe._dict({"from_date": now(), "to_date": add_to_date(now(), months = 1)}))
+
+		item_map = get_item_map(rows)
+
+		self.assertEquals(item_map['Test_One_Plus'][11],6789.0)
+		self.assertEquals(item_map['Test_Mac_Book_Pro'][11],12345.0)
+
+	def test_for_stock_balance_with_warehouse_and_warehouse_type(self):
+
+		item_1 = get_item("Test_item_for_warehouse_type", 2367)
+		wt = get_warehouse_type()
+		item_1.reload()
+		warehouse = frappe.get_doc("Warehouse", item_1.item_defaults[0].default_warehouse)
+		warehouse.warehouse_type = wt.name
+		warehouse.save()
+
+		cols, rows = execute(filters=frappe._dict({
+				"from_date": now(),
+				"to_date": add_to_date(now(), months = 1),
+				"warehouse_type": wt.name
+			}))
+
+		item_map = get_item_map(rows)
+
+		self.assertEquals(item_map['Test_item_for_warehouse_type'][5], item_1.item_defaults[0].default_warehouse)
+		self.assertEquals(item_map['Test_item_for_warehouse_type'][11], 2367)
+
+def get_item_map(rows):
+
+	item_map = {}
+	for row in rows:
+		item_map[row[0]] = row
+
+	return item_map
+
+
+def get_item(item_code, qty):
```

Comment:
repetitive function (?)

Reply:
I just checked there is only one method for creating the item map.